hpeOneView/resources/servers/server_hardware.py

```diff
@@ -326,3 +333,68 @@ def get_local_storage(self, ip=None):
             uri = "{}?ip={}".format(uri, ip)
 
         return self._helper.do_get(uri)
+
+    def check_firmware_compliance(self, configuration, timeout=-1, custom_headers=None):
+        """
+        Checks the firmware compliance of a server with the selected firmware baseline.
+        Returns the compliance status of each individual component as well as the overall
+        firmware compliance status of the server.
+
+        Args:
+            configuration: Firmware Compliance Configuration
+
+        Returns:
+            Resource
+        """
+        uri = "{}/firmware-compliance".format(self.URI)
+        return self._helper.do_post(uri, configuration, timeout=timeout, custom_headers=custom_headers)
+
+    @ensure_resource_client
+    def validate_server_hardware_for_firmware_update(self):
+        """
+        Performs a validation for the prerequisites before firmware update.
+        Checks if the server hardware is in powered off state, No server profile attached to server
+        hardware, server hardware model is Gen10 or above, server hardware has advanced ilo license type.
+
+        """
+        state = self.data["state"]
+        server_profile_uri = self.data["serverProfileUri"]
+        power_state = self.data["powerState"]
+        generation = self.data["mpModel"]
+        license_type = self.data["mpLicenseType"]
+
+        validation_error_list = []
+
+        matches = re.findall("\d+$", generation)
+        if int(matches[0]) < 5:
```

Comment:
How it is checking the generation here?

Reply:
generation is checked from mpModel. Gen10 is of iLO5 or more.